extensions/kubernetes-config/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesConfigRecorder.java

```diff
@@ -26,6 +26,17 @@
                 KubernetesClientUtils.createClient(clientConfig)));
     }
 
+    public void warnAboutSecrets(KubernetesConfigSourceConfig config, KubernetesConfigBuildTimeConfig buildTimeConfig) {
+        if (config.enabled
+                && config.secrets.isPresent()
+                && !config.secrets.get().isEmpty()
+                && !buildTimeConfig.secretsEnabled) {
+            log.warn("Configuration is read from Secrets " + config.secrets.get()
+                    + ", but quarkus.kubernetes-config.secrets.enabled is false."
+                    + " Check if your application's service account has enough permissions to read secrets.");
```

Comment:
Is it just me, or does the last sentence seem out of place?

Reply:
I wanted to say that what the user ultimately needs to do is _not_ setting our config property, but make sure that the app's service account has access to secrets.

Obviously `quarkus.kubernetes-config.secrets.enabled` is the easiest way, if people use the Kubernetes extension. But that doesn't always have to be the case.

Reply:
Okay